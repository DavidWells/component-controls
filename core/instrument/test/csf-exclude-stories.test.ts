import { loadStoriesTests } from './loadTestFiles';

describe('csf-exclude-stories', () => {
  loadStoriesTests(
    { stories: { storeSourceFile: false } },
    'csf',
    'exclude-stories',
  );
});
