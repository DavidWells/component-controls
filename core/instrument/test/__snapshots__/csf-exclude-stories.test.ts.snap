// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csf-exclude-stories exclude-stories-array.js 1`] = `
Object {
  "components": Object {},
  "kinds": Object {
    "Storybook/Blocks/EXCLUDEStoryArray": Object {
      "components": Object {},
      "excludeStories": Array [
        "story1",
        "story2",
      ],
      "package": "7b45564d8ad9cf0f6da12c3360f96029",
      "title": "Storybook/Blocks/EXCLUDEStoryArray",
    },
  },
  "packages": Object {
    "7b45564d8ad9cf0f6da12c3360f96029": Object {
      "fileHash": "7b45564d8ad9cf0f6da12c3360f96029",
      "name": "@component-controls/instrument",
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/exclude-stories/exclude-stories-array.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.0.1",
    },
  },
  "stories": Object {
    "story3": Object {
      "arguments": Array [],
      "id": "story3",
      "loc": Object {
        "end": Object {
          "column": 30,
          "line": 8,
        },
        "start": Object {
          "column": 22,
          "line": 8,
        },
      },
      "name": "story3",
      "source": "() => {}",
    },
  },
  "transformed": "export default {
  title: 'Storybook/Blocks/EXCLUDEStoryArray',
  excludeStories: ['story1', 'story2'],
};

export const story1 = () => {};
export const story2 = () => {};
export const story3 = () => {};
",
}
`;

exports[`csf-exclude-stories exclude-stories-regex.js 1`] = `
Object {
  "components": Object {},
  "kinds": Object {
    "Storybook/Blocks/ExcludeStoryRegex": Object {
      "components": Object {},
      "excludeStories": "story(2|3)",
      "package": "7b45564d8ad9cf0f6da12c3360f96029",
      "title": "Storybook/Blocks/ExcludeStoryRegex",
    },
  },
  "packages": Object {
    "7b45564d8ad9cf0f6da12c3360f96029": Object {
      "fileHash": "7b45564d8ad9cf0f6da12c3360f96029",
      "name": "@component-controls/instrument",
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/exclude-stories/exclude-stories-regex.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.0.1",
    },
  },
  "stories": Object {
    "story1": Object {
      "arguments": Array [],
      "id": "story1",
      "loc": Object {
        "end": Object {
          "column": 30,
          "line": 6,
        },
        "start": Object {
          "column": 22,
          "line": 6,
        },
      },
      "name": "story1",
      "source": "() => {}",
    },
  },
  "transformed": "export default {
  title: 'Storybook/Blocks/ExcludeStoryRegex',
  excludeStories: /story(2|3)/,
};

export const story1 = () => {};
export const story2 = () => {};
export const story3 = () => {};
",
}
`;

exports[`csf-exclude-stories include-stories-array.js 1`] = `
Object {
  "components": Object {},
  "kinds": Object {
    "Storybook/Blocks/IncludeStoryArray": Object {
      "components": Object {},
      "includeStories": Array [
        "story1",
        "story2",
      ],
      "package": "7b45564d8ad9cf0f6da12c3360f96029",
      "title": "Storybook/Blocks/IncludeStoryArray",
    },
  },
  "packages": Object {
    "7b45564d8ad9cf0f6da12c3360f96029": Object {
      "fileHash": "7b45564d8ad9cf0f6da12c3360f96029",
      "name": "@component-controls/instrument",
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/exclude-stories/include-stories-array.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.0.1",
    },
  },
  "stories": Object {
    "story1": Object {
      "arguments": Array [],
      "id": "story1",
      "loc": Object {
        "end": Object {
          "column": 30,
          "line": 6,
        },
        "start": Object {
          "column": 22,
          "line": 6,
        },
      },
      "name": "story1",
      "source": "() => {}",
    },
    "story2": Object {
      "arguments": Array [],
      "id": "story2",
      "loc": Object {
        "end": Object {
          "column": 30,
          "line": 7,
        },
        "start": Object {
          "column": 22,
          "line": 7,
        },
      },
      "name": "story2",
      "source": "() => {}",
    },
  },
  "transformed": "export default {
  title: 'Storybook/Blocks/IncludeStoryArray',
  includeStories: ['story1', 'story2'],
};

export const story1 = () => {};
export const story2 = () => {};
export const story3 = () => {};
",
}
`;

exports[`csf-exclude-stories include-stories-regex.js 1`] = `
Object {
  "components": Object {},
  "kinds": Object {
    "Storybook/Blocks/IncludeStoryRegex": Object {
      "components": Object {},
      "includeStories": "story(2|3)",
      "package": "7b45564d8ad9cf0f6da12c3360f96029",
      "title": "Storybook/Blocks/IncludeStoryRegex",
    },
  },
  "packages": Object {
    "7b45564d8ad9cf0f6da12c3360f96029": Object {
      "fileHash": "7b45564d8ad9cf0f6da12c3360f96029",
      "name": "@component-controls/instrument",
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/exclude-stories/include-stories-regex.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.0.1",
    },
  },
  "stories": Object {
    "story2": Object {
      "arguments": Array [],
      "id": "story2",
      "loc": Object {
        "end": Object {
          "column": 30,
          "line": 7,
        },
        "start": Object {
          "column": 22,
          "line": 7,
        },
      },
      "name": "story2",
      "source": "() => {}",
    },
    "story3": Object {
      "arguments": Array [],
      "id": "story3",
      "loc": Object {
        "end": Object {
          "column": 30,
          "line": 8,
        },
        "start": Object {
          "column": 22,
          "line": 8,
        },
      },
      "name": "story3",
      "source": "() => {}",
    },
  },
  "transformed": "export default {
  title: 'Storybook/Blocks/IncludeStoryRegex',
  includeStories: /story(2|3)/,
};

export const story1 = () => {};
export const story2 = () => {};
export const story3 = () => {};
",
}
`;
